src/V3File.cpp

```diff
@@ -916,19 +917,142 @@ void V3OutFormatter::printf(const char* fmt...) {
+        auto iter = s_freeBuffers.begin();
+        s_buffersInUse.splice(s_buffersInUse.end(), s_freeBuffers, iter);
+        iter->clear();
+        return AioWriteBufferHandle(iter);
+    }
+
+    aiocb* aiocbpArray[s_buffersInKernel.size()];
+    auto queuedBufferIter = s_buffersInKernel.begin();
+    for (auto& cbp : aiocbpArray) {
+        cbp = &queuedBufferIter->m_aiocb;
+        ++queuedBufferIter;
+    }
+    // Wait for any queued buffer to complete.
+    aioSuspendIgnoreInterrupts(aiocbpArray, s_buffersInKernel.size());
+
+    // Find first buffer that completed.
+    auto iter = s_buffersInKernel.begin();
+    for (; iter != s_buffersInKernel.end(); ++iter) {
+        if (aioOperationHasCompleted(iter->m_aiocb)) break;
+    }
+    UASSERT((iter != s_buffersInKernel.end()), "aio_suspend returned without available buffer");
+    s_buffersInUse.splice(s_buffersInUse.end(), s_buffersInKernel, iter);
+    iter->clear();
+    return AioWriteBufferHandle(iter);
+}
+
+void AioWriter::queueBuffer(AioWriteBufferHandle handle) {
+    auto iter = handle.m_iter;
+    if (iter->size() == 0) {
+        s_freeBuffers.splice(s_freeBuffers.end(), s_buffersInUse, iter);
+        return;
+    }
+    s_buffersInKernel.splice(s_buffersInKernel.end(), s_buffersInUse, iter);
+    UASSERT((iter->m_aiocb.aio_fildes >= 0), "Buffer queued with invalid FD");
+    if (aio_write(&iter->m_aiocb) < 0)
+        v3fatal("Asynchronous write failure: " << std::strerror(errno));
+}
+
+void AioWriter::waitForCompletion() {
+    aiocb* aiocbpArray[s_buffersInKernel.size()];
+    aiocb** startp = &aiocbpArray[0];
+    aiocb** endp = startp;
+
+    for (auto& buf : s_buffersInKernel) { *endp++ = &buf.m_aiocb; }
+
+    waitForCompletionInternal(startp, endp);
+    s_freeBuffers.splice(s_freeBuffers.end(), s_buffersInKernel);
+}
+
+void AioWriter::waitForCompletion(int fd) {
+    std::list<AioWriteBuffer> bufferList;
+    aiocb* aiocbpArray[s_buffersInKernel.size()];
+    aiocb** startp = &aiocbpArray[0];
+    aiocb** endp = startp;
+
+    auto iter = s_buffersInKernel.begin();
+    while (iter != s_buffersInKernel.end()) {
+        const auto next = std::next(iter);
+        if (iter->m_aiocb.aio_fildes == fd) {
+            *endp++ = &iter->m_aiocb;
+            bufferList.splice(bufferList.end(), s_buffersInKernel, iter);
+        }
+        iter = next;
+    }
+    waitForCompletionInternal(startp, endp);
+    s_freeBuffers.splice(s_freeBuffers.end(), bufferList);
+}
+
 //######################################################################
 // V3OutFormatter: A class for printing to a file, with automatic indentation of C++ code.
 
 V3OutFile::V3OutFile(const string& filename, V3OutFormatter::Language lang)
-    : V3OutFormatter{filename, lang} {
-    if ((m_fp = V3File::new_fopen_w(filename)) == nullptr) {
-        v3fatal("Cannot write " << filename);
+    : V3OutFormatter{filename, lang}
+    , m_offset(0)
+    , m_bufferHandle(AioWriter::dequeueBuffer()) {
```

Comment:
```suggestion
    : V3OutFormatter{filename, lang}
    , m_offset{0}
    , m_bufferHandle{AioWriter::dequeueBuffer()} {
```

though m_offset is better set = 0 where it is declared.